homeassistant/components/deconz/button.py

```diff
@@ -0,0 +1,115 @@
+"""Support for deCONZ buttons."""
+
+from __future__ import annotations
+
+from collections.abc import ValuesView
+from dataclasses import dataclass
+
+from pydeconz.group import Scene as PydeconzScene
+
+from homeassistant.components.button import (
+    DOMAIN,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .deconz_device import DeconzSceneMixin
+from .gateway import DeconzGateway, get_gateway_from_config_entry
+
+
+@dataclass
+class DeconzButtonDescriptionMixin:
+    """Required values when describing deCONZ button entities."""
+
+    suffix: str
+    button_fn: str
+
+
+@dataclass
+class DeconzButtonDescription(ButtonEntityDescription, DeconzButtonDescriptionMixin):
+    """Class describing deCONZ button entities."""
+
+
+ENTITY_DESCRIPTIONS = {
+    PydeconzScene: [
+        DeconzButtonDescription(
+            key="store",
+            button_fn="store",
+            suffix="Store Current Scene",
+            icon="mdi:inbox-arrow-down",
+            entity_category=EntityCategory.CONFIG,
+        )
+    ]
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the deCONZ button entity."""
+    gateway = get_gateway_from_config_entry(hass, config_entry)
+    gateway.entities[DOMAIN] = set()
+
+    @callback
+    def async_add_scene(
+        scenes: list[PydeconzScene]
+        | ValuesView[PydeconzScene] = gateway.api.scenes.values(),
+    ) -> None:
+        """Add scene button from deCONZ."""
+        entities = []
+
+        for scene in scenes:
+
+            known_entities = set(gateway.entities[DOMAIN])
+            for description in ENTITY_DESCRIPTIONS.get(PydeconzScene, []):
+
+                new_entity = DeconzButton(scene, gateway, description)
+                if new_entity.unique_id not in known_entities:
```

Comment:
Is would be a bit more efficient if you could avoid creating the entity object until you know the unique id is not in known entities.

Reply:
Definitely, Im reworking stuff in the library as I want to remove signals to here which are not explicitly new devices to avoid this